### PR TITLE
Version 1

### PR DESCRIPTION
**Pull request by orta**
Changes from yours:
- Uses `pod lib create` template for project, which means free carthage support https://github.com/floriankugler/FLKAutoLayout/pull/32
- Contains fixes to https://github.com/floriankugler/FLKAutoLayout/pull/43 https://github.com/floriankugler/FLKAutoLayout/pull/34 and  https://github.com/floriankugler/FLKAutoLayout/pull/28 and https://github.com/floriankugler/FLKAutoLayout/pull/26

I mainly did 3 things: 
- forced `nonnull` on predicates, our biggest source of AL bugs
- tightened up the types to be lightweight generics, and made anything that really returns one predicate only return that predicate in the public API
- added comments throughout the public API

Contains #1 too - and as a part of cleaning up the API I switched the code style so I could see what I've touched so far

### CODE REVIEW

File: FLKAutoLayout/Private/FLKAutoLayoutPredicateList.m [hunk 1]
```diff
@@ -12,13 +12,13 @@ @implementation FLKAutoLayoutPredicateList {
     NSMutableArray* predicates;
 }
 
-+ predicateListFromString:(NSString*)string {
++ predicateListFromString:(NSString *)string {
     FLKAutoLayoutPredicateList* predicateList = [[FLKAutoLayoutPredicateList alloc] init];
-    NSArray* predicateStrings = [string componentsSeparatedByString:@","];
+    NSArray * predicateStrings = [string componentsSeparatedByString:@","];
```
Comment (alloy) [inline FLKAutoLayout/Private/FLKAutoLayoutPredicateList.m]: Contains extra space after asterisk.

File: FLKAutoLayout/Private/FLKAutoLayoutPredicateList.m [hunk 2]
```diff
@@ -34,7 +34,7 @@ - (id)init {
     return self;
 }
 
-- (NSArray*)iteratePredicatesUsingBlock:(predicateBlock)block {
+- (NSArray *)iteratePredicatesUsingBlock:(predicateBlock)block {
     NSMutableArray* constraints = [NSMutableArray array];
```
Comment (alloy) [inline FLKAutoLayout/Private/FLKAutoLayoutPredicateList.m]: Still has asterisk to the left.

File: FLKAutoLayout/UIView+FLKAutoLayout.h [hunk 3]
```diff
@@ -1,74 +1,157 @@
...
-//
+@import UIKit;
+NS_ASSUME_NONNULL_BEGIN
 
+FOUNDATION_EXTERN NSString * const FLKNoConstraint;
 
-#import <UIKit/UIKit.h>
-
+#define LayoutConstraintArray NSArray <NSLayoutConstraint *>
+#define ViewArray NSArray <UIView *>
```
Comment (alloy) [inline FLKAutoLayout/UIView+FLKAutoLayout.h]: Does this show-up correctly in Xcode’s inline/adhoc doc viewer?

**Comment by alloy** [comment]
> forced `nonnull` on predicates, our biggest source of AL bugs
> …
> and made anything that really returns one predicate only return that predicate in the public API

Ace 👍

I did a write-up somewhere about what to do wrt semantic versioning, but can’t find it anymore. Anyways, this clearly needs a major bump, right?

**Comment by alloy** [comment]
The asterisks are still messy in many more locations, didn’t feel like pointing them all out. Was that something you had wanted to all fix or was it just accidentally in a few places?

**Comment by alloy** [comment]
And FYI, changing whitespace rules makes a fork way more definitive, which is why I had not done it yet before. I’m not opposed to forking, though.

**Comment by orta** [comment]
> The asterisks are still messy in many more locations, didn’t feel like pointing them all out. Was that something you had wanted to all fix or was it just accidentally in a few places?

More accidental, I only really cared about it in the public API

> And FYI, changing whitespace rules makes a fork way more definitive, which is why I had not done it yet before. I’m not opposed to forking, though.

That's fair, Florian is probably pretty chill about it

> I did a write-up somewhere about what to do wrt semantic versioning, but can’t find it anymore. Anyways, this clearly needs a major bump, right?

Aye it's gone from `0.2` -> `1.0`

**Comment by orta** [comment]
OK, I've made all the files have a consistent license-ish header thing, the whitespacing should be a lot more consistent now too.

**Comment by alloy** [comment]
Re the license, if Florian approves, I’d prefer to remove all source file headers and just have a LICENSE file that shows the proper copyright lines for each of us.

**Comment by alloy** [comment]
Also could do with some updating of the README.

**Comment by orta** [comment]
Looks like mainly removing the `nil`s and adding the UIViewController top/bottom guides

**Comment by alloy** [comment]
Aye

**Pull request merged**